src/CoreBundle/DataAccess/MenuItemDataAccess.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+/*
+ * This file is part of the Chameleon System (https://www.chameleonsystem.com).
+ *
+ * (c) ESONO AG (https://www.esono.de)
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace ChameleonSystem\CoreBundle\DataAccess;
+
+use ChameleonSystem\CoreBundle\Bridge\Chameleon\Module\Sidebar\MenuCategory;
+use ChameleonSystem\CoreBundle\Bridge\Chameleon\Module\Sidebar\MenuItem;
+use ChameleonSystem\CoreBundle\Bridge\Chameleon\Module\Sidebar\MenuItemFactoryInterface;
+use ChameleonSystem\CoreBundle\DataModel\MenuCategoryAndItem;
+
+class MenuItemDataAccess implements MenuItemDataAccessInterface
+{
+    /**
+     * @var MenuItemFactoryInterface
+     */
+    private $menuItemFactory;
+
+    public function __construct(MenuItemFactoryInterface $menuItemFactory)
+    {
+        $this->menuItemFactory = $menuItemFactory;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function getMenuCategories(): array
+    {
+        $activeUser = \TCMSUser::GetActiveUser();
+        if (null === $activeUser) {
+            return [];
+        }
+
+        $menuCategories = [];
+
+        $tdbCategoryList = \TdbCmsMenuCategoryList::GetList();
+        $tdbCategoryList->ChangeOrderBy([
+            '`cms_menu_category`.`position`' => 'ASC',
+        ]);
+        $tdbCategoryList->GoToStart();
+        while (false !== $tdbCategory = $tdbCategoryList->Next()) {
```

Comment:
```suggestion
        while ($tdbCategory = $tdbCategoryList->Next()) {
```
The false check is not necessary in this loop. The loop will end at the last item of $tdbCategoryList

Reply:
I converted it into a classic PHP for-each-loop, which is now possible.